### PR TITLE
fix: improve performance of npm_package with removal of .to_list() on additional_files

### PR DESCRIPTION
**Pull request by gregmagolan**
I was hoping to re-use the docstring from attributes in copy_to_directory but as a macro the docstring must be a string literal so no string operations are allowed to build it

### CODE REVIEW

File: npm/private/npm_package.bzl [hunk 1]
```diff
@@ -16,201 +16,72 @@ load("//js:libs.bzl", "js_lib_helpers")
...
-and adding them to the list of files available to copy can add noticeable overhead to the analysis phase in a large
-repository with many npm_package targets.
-""",
-        # TODO(2.0): flip default to False
-        default = True,
-    ),
+_NPM_PACKAGE_ATTRS = dicts.add(copy_to_directory_lib_attrs, {
+    "package": attr.string(),
+    "version": attr.string(),
+    "data": attr.label_list(),
```
Comment (kormide) [inline npm/private/npm_package.bzl:26]: nit: sort these attribute dicts alphabetically?

Comment (gregmagolan) [reply npm/private/npm_package.bzl:26]: ✅

**Review by kormide — APPROVED** [review]

File: npm/private/npm_package.bzl [hunk 2]
```diff
@@ -256,19 +127,340 @@ def _impl(ctx):
...
+            include_declarations = include_declarations,
+            include_transitive_declarations = include_transitive_declarations,
+            include_runfiles = include_runfiles,
+            # Always tag the target manual since we should only build it when the final target is built.
+            tags = kwargs.get("tags", []) + ["manual"],
+            # Always propagate the testonly attribute
+            testonly = kwargs.get("testonly", False),
+        )
+
+    if files_target:
```
Comment (jbedard) [inline npm/private/npm_package.bzl:442]: Can't this just merge with the `if` above?

Comment (gregmagolan) [reply npm/private/npm_package.bzl:442]: Hmm. Seems so. The code path changed and I didn't notice the duplicate

**Pull request merged**